linux/Platform.c

```diff
@@ -854,3 +859,134 @@ void Platform_getBattery(double* percent, ACPresence* isOnAC) {
    Platform_Battery_cacheIsOnAC = *isOnAC;
    Platform_Battery_cacheTime = now;
 }
+
+char** Platform_getLocalIPv4addressChoices() {
+   int s = socket(AF_INET, SOCK_STREAM, 0);
+   if (s < 0)
+      return NULL;
+
+   struct ifconf ifconf;
+   char ifBuffer[4096];
+   ifconf.ifc_buf = ifBuffer;
+   ifconf.ifc_len = sizeof(ifBuffer);
+
+   if (ioctl(s, SIOCGIFCONF, &ifconf) == -1) {
+      close(s);
+      return NULL;
+   }
+
+   close(s);
+
+   const struct ifreq *ifr = (const struct ifreq *)ifBuffer;
+   unsigned int count = ifconf.ifc_len / sizeof(*ifr);
+
+   char** ret = xMalloc((count + 1) * sizeof(char**));
```

Comment:
The `sizeof`'d type has one to much indirection:
```suggestion
   char** ret = xMalloc((count + 1) * sizeof(char*));
```